src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,48 @@
 import React from 'react';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from '../../app/hooks';
+import { Filter, filterActions } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useDispatch();
+  const { query, status } = useAppSelector((state) => state.filter);
+
+  const handleFilterChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedFilter = event.target.value;
+
+    if (selectedFilter === 'all') {
```

Comment:
```suggestion
    if (selectedFilter === Filter.ALL) {
```
you should use enum values everywhere instead of strings

Also, it would be better to use `switch` block here